### ISSUE TITLE
Expose icon names in `core/src/icon/index.ts`

### ISSUE THREAD

**Issue by astorije** [vmware/clarity#5335]
## Summary

**What is the change?**

Re-exporting icon names (e.g. `infoStandardIconName`) in `core/src/icon/index.ts` and not only `@clr/core/icon/shapes/<icon>`

**Why should it go in Clarity?**

When working in TypeScript, since the `shape` prop of the `CdsIcon` component is typed as a `string`, it's very easy to make a typo that goes unchecked by `tsc`, the test runner, etc. For example `<CdsIcon shape="info-standand" />`

There are 2 ways this could be addressed by the library, on the top of my head:

- Change the `shape` prop type from `string` to a union type: `... | 'info-standard' | ...`
- Exposing icon names and recommending their usage: `<CdsIcon shape={infoStandardIconName} />`

Both options can be combined for maximum type safety.
The first option is how Font Awesome works (EDIT: I forgot that Font Awesome _also_ supports the second option), for instance, and it has been helpful to us many times. But it's a bigger change in the library, and somewhat a breaking change (but if it breaks one's app, it was probably a typo). More importantly, it does not ensure that the `info-standard` icon was actually loaded by `ClarityIcons.addIcons()` so, while safer at compile time, it might still silently break at runtime.
The second option would be trivial to support here since you already expose the icon names in the `shapes` files, it would just be a bunch of extra exports in `icon/index.ts`. And it would also be completely backwards compatible, just an extra line [in the doc](https://clarity.design/storybook/core/?path=/story/components-icon-getting-started--page) to recommend TypeScript users to optionally use the icon name instead of the string.

I would be more than happy to open a PR if you think this is a good addition to the library.

## Examples

```tsx
import { ClarityIcons, infoStandardIcon, infoStandardIconName } from '@clr/core/icon';
import { CdsIcon } from '@clr/react/icon';

ClarityIcons.addIcons(infoStandardIcon);

<CdsIcon shape={infoStandardIconName} />
```

## Workarounds

At the moment, it can be done by loading the `shapes` files, but it's extra (and undocumented) boilerplate that does not make it very discoverable:

```tsx
import { ClarityIcons, infoStandardIcon } from '@clr/core/icon';
import { infoStandardIconName } from '@clr/core/icon/shapes/info-standard';
import { CdsIcon } from '@clr/react/icon';

ClarityIcons.addIcons(infoStandardIcon);

<CdsIcon shape={infoStandardIconName} />
```

**Comment by bbogdanov** [comment]
Hi @astorije,

Thank you for the awesome proposal. I believe it is doable and it will definitely help with future development.

@mathisscott Can you please take a look at this one?

**Comment by astorije** [comment]
Thanks for your feedback, @bbogdanov!
I have opened a PR at https://github.com/vmware/clarity/pull/5337. ~I am currently looking at adding a blob to the documentation but figured I would open a PR as early as possible.~ EDIT: Done, though open to suggestions!

**Comment by bbogdanov** [comment]
@astorije Thank you once again for the great contribution!

**Comment by astorije** [comment]
My pleasure! Do you have any sense of when it will be available in a release?

**Comment by mathisscott** [comment]
@astorije 
It's merged into next so it will be released with the next 5.0 release

**Comment by astorije** [comment]
Perfect, thank you!
